services/src/main/scala/cromwell/services/MetadataQuery.scala

```diff
@@ -1,28 +1,39 @@
 package cromwell.services
 
 import java.sql.Timestamp
+import java.time.OffsetDateTime
 
-import cromwell.backend.BackendJobDescriptorKey
 import cromwell.core.WorkflowId
-import cromwell.engine.ExecutionIndex._
 
-
-case class MetadataJobKey(callFqn: String, index: ExecutionIndex, attempt: Int)
-object MetadataJobKey {
-  def apply(backendJobKey: BackendJobDescriptorKey): MetadataJobKey = MetadataJobKey(backendJobKey.call.fullyQualifiedName, backendJobKey.index, backendJobKey.attempt)
-}
+case class MetadataJobKey(callFqn: String, index: Option[Int], attempt: Int)
 
 case class MetadataKey(workflowId: WorkflowId, jobKey: Option[MetadataJobKey], key: String)
+
 case class MetadataValue(value: String)
+
+object MetadataEvent {
+  def apply(key: MetadataKey, value: MetadataValue) = {
+    // TODO: Look into db/slick using OffsetDateTime, or storing datetimes as UTC?
```

Comment:
So far every confab I've partaken in since you commented on that one ticket has been positive. How about after standup tomorrow we discuss briefly to make sure there's no disagreement and make a ticket for this?